pkg/skaffold/initializer/init.go

```diff
@@ -150,94 +172,136 @@ func DoInit(out io.Writer, c Config) error {
 	return nil
 }
 
-func processCliArtifacts(artifacts []string) ([]dockerfilePair, error) {
-	var pairs []dockerfilePair
+// autoSelectBuilders takes a list of builders and images, checks if any of the builders' configured target
+// images match an image in the image list, and returns a list of the matching builder/image pairs. Also
+// returns the images from the original image list that didn't match any build configurations.
+func autoSelectBuilders(buildConfigs []InitBuilder, images []string) ([]BuilderImagePair, []string) {
+	// Auto-select builders that have a definite target image
+	pairs := []BuilderImagePair{}
+	filteredImages := []string{}
+	for _, image := range images {
+		matchingConfigIndex := -1
+		for i, config := range buildConfigs {
+			if image != config.GetConfiguredImage() {
+				continue
+			}
+
+			// Found more than one match; can't auto-select.
+			if matchingConfigIndex != -1 {
+				matchingConfigIndex = -1
+				break
+			}
+			matchingConfigIndex = i
+		}
+
+		if matchingConfigIndex != -1 {
+			// Exactly one pair found; save the pair and remove from remaining build configs
+			pairs = append(pairs, BuilderImagePair{ImageName: image, Builder: buildConfigs[matchingConfigIndex]})
+			buildConfigs = append(buildConfigs[:matchingConfigIndex], buildConfigs[matchingConfigIndex+1:]...)
+		} else {
+			// No definite pair found, add to images list
+			filteredImages = append(filteredImages, image)
+		}
+	}
+	return pairs, filteredImages
+}
+
+func detectBuildFile(path string) ([]InitBuilder, error) {
```

Comment:
Sorry, another followup thing. Since Jib can return multiple build configs from a single file (i.e. parent pom in a multimodule project) we need to return a list here.